src/rdf4cpp/rdf/Namespace.hpp

```diff
@@ -31,7 +32,7 @@ class Namespace {
     /**
      * NodeStorage from which IRI objects are created.
      */
-    mutable NodeStorage node_storage_;
+    WeakNodeStorage node_storage_;
```

Comment:
Here I would say that a WeakNodeStorage is the right choice. Otherwise it get's ugly with the default instances.